cmd/autopublicizeconfig/main.go

```diff
@@ -0,0 +1,207 @@
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.GitHubOptions.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Failed to start secrets agent")
+	}
+
+	gc, err := o.GitHubOptions.GitHubClient(sa, o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("error getting GitHub client")
+	}
+
+	logrus.Info("Getting repositories that exists in the whitelist or promote official images")
+	orgRepos, err := getReposForPrivateOrg(o.releaseRepoPath, o.WhitelistOptions.WhitelistConfig.Whitelist)
+	if err != nil {
+		logrus.WithError(err).Fatal("couldn't get the list of org/repos that promote official images")
+	}
+
+	publicizeRepos := make(map[string]string)
+	for org, repos := range orgRepos {
+		for repo := range repos {
+			privateOrgRepo := fmt.Sprintf("%s/%s", privateOrg, repo)
+			publicOrgRepo := fmt.Sprintf("%s/%s", org, repo)
+			publicizeRepos[privateOrgRepo] = publicOrgRepo
+		}
+	}
+
+	config := &Config{}
+	config.Repositories = publicizeRepos
+
+	b, err := yaml.Marshal(&config)
+	if err != nil {
+		logrus.WithError(err).Fatal("couldn't marshal the publicize configuration")
+	}
+
+	logrus.Info("Generating publicize config file")
+	if err := os.MkdirAll(path.Dir(o.publicizeConfig), os.ModePerm); err != nil && !os.IsExist(err) {
+		logrus.WithError(err).Fatal("failed to ensure directory existed for new publicize configuration")
+	}
+	if err := ioutil.WriteFile(o.publicizeConfig, b, 0664); err != nil {
+		logrus.WithError(err).Fatal("failed to write new publicize configuration")
+	}
+
+	stdout := bumper.HideSecretsWriter{Delegate: os.Stdout, Censor: sa}
+	stderr := bumper.HideSecretsWriter{Delegate: os.Stderr, Censor: sa}
+
+	changed, err := bumper.HasChanges()
+	if err != nil {
+		logrus.WithError(err).Fatal("error occurred when checking changes")
+	}
+
+	if !changed {
+		logrus.Info("No changes to commit")
+		os.Exit(0)
+	}
+
+	logrus.Info("Preparing pull request")
+	title := fmt.Sprintf("%s %s", matchTitle, time.Now().Format(time.RFC1123))
+	if err := bumper.GitCommitAndPush(fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", o.githubLogin, string(sa.GetTokenGenerator(o.GitHubOptions.TokenPath)()), o.githubLogin, githubRepo), remoteBranch, o.gitName, o.gitEmail, title, stdout, stderr); err != nil {
+		logrus.WithError(err).Fatal("Failed to push changes.")
+	}
+
+	var labelsToAdd []string
+	if o.selfApprove {
+		logrus.Infof("Self-aproving PR by adding the %q and %q labels", labels.Approved, labels.LGTM)
+		labelsToAdd = append(labelsToAdd, labels.Approved, labels.LGTM)
+	}
+
+	source := fmt.Sprintf("%s:%s", o.githubLogin, remoteBranch)
+	if err := bumper.UpdatePullRequestWithLabels(gc, githubOrg, githubRepo, title, description, matchTitle, source, "master", true, labelsToAdd); err != nil {
+		logrus.WithError(err).Fatal("PR creation failed.")
+	}
+}
+
+func getReposForPrivateOrg(releaseRepoPath string, whitelist map[string][]string) (map[string]sets.String, error) {
```

Comment:
s/whitelist/allowlist/g

Reply:
this is a term that we are already using. see also https://github.com/openshift/ci-tools/blob/master/pkg/config/whitelist.go

Reply:
https://www.redhat.com/en/blog/making-open-source-more-inclusive-eradicating-problematic-language

Feel free to also change the terminology in that package (in a different PR)

Reply:
Oh!!! Thanks for letting me know. I will make the changes